src/gist.rs

```diff
@@ -109,6 +114,28 @@ impl GistClient {
         Self::get_response(response)
     }
 
+    fn gist_id_from_url<'a>(&self, gist_url: &'a str) -> Option<&'a str> {
+        let re = Regex::new(
+            // Expect URL like https://gist.github.com/<user>/<gist_id>
+            r"https://gist\.github\.com/.+/(?P<gist_id>[0-9a-f]+)$",
```

Comment:
If only the gist_id is needed to fetch then maybe it makes more sense for the user to input the ID directly instead of the full URL?

Reply:
I chose URL because I was thinking most of the time you probably find a Gist via Google and its slightly less clicks to copy the whole URL than trying to select the ID (last part of it).

Reply:
Okay, makes sense, I agree that fewer clicks is better